app/controllers/declaration/NatureOfTransactionController.scala

```diff
@@ -55,10 +55,16 @@ class NatureOfTransactionController @Inject()(
     form().bindFromRequest
       .fold(
         (formWithErrors: Form[NatureOfTransaction]) => Future.successful(BadRequest(natureOfTransactionPage(mode, formWithErrors))),
-        form => updateCache(form).map(_ => navigator.continueTo(mode, controllers.declaration.routes.PreviousDocumentsController.displayPage))
+        form => updateCache(form).map(_ => redirect(mode))
       )
   }
 
   private def updateCache(formData: NatureOfTransaction)(implicit r: JourneyRequest[AnyContent]): Future[Option[ExportsDeclaration]] =
     updateExportsDeclarationSyncDirect(model => model.copy(natureOfTransaction = Some(formData)))
+
+  private def redirect(mode: Mode)(implicit request: JourneyRequest[AnyContent]) =
+    request.cacheModel.previousDocuments.map(_.documents).getOrElse(Seq.empty) match {
+      case Seq() => navigator.continueTo(mode, controllers.declaration.routes.PreviousDocumentsController.displayPage)
```

Comment:
Isn't the PreviousDocumentsSummaryController doing this logic already?  I.e. if there are no cached prev. docs, redirect to "add" page.

Reply:
So this controller just needs to direct to PreviousDocumentsSummaryController.

Reply:
Done, good points with all comments about redirections